### ISSUE TITLE
Nomad 1.5.1 client unable to start with IPv6 server addresses 

### ISSUE THREAD

**Issue by protochron** [hashicorp/nomad#16586]
### Nomad version
Nomad v1.5.1
BuildDate 2023-03-10T22:05:57Z
Revision 6c118ddaf95e9bfd3d85f488f255fdc5e14129c7

### Operating system and Environment details
arm64 LInux

### Issue
An agent attempting to join a cluster that only advertises addresses with IPv6 with Consul is broken because the IP address is improperly formatted. This may be related to #16550.

Reverting to nomad 1.5.0 fixes the issue.

Here's the output I'm getting in the logs when the agent starts (truncated for brevity):
```
Mar 21 14:20:00 ip-10-10-20-173 nomad[76334]:     2023-03-21T14:20:00.118Z [WARN]  client.fingerprint_mgr.network: unable to parse speed: path=/usr/sbin/e>
Mar 21 14:20:00 ip-10-10-20-173 nomad[76334]:     2023-03-21T14:20:00.200Z [INFO]  client.fingerprint_mgr.vault: Vault is available
Mar 21 14:20:00 ip-10-10-20-173 nomad[76334]:     2023-03-21T14:20:00.215Z [INFO]  client.plugin: starting plugin manager: plugin-type=csi
Mar 21 14:20:00 ip-10-10-20-173 nomad[76334]:     2023-03-21T14:20:00.215Z [INFO]  client.plugin: starting plugin manager: plugin-type=driver
Mar 21 14:20:00 ip-10-10-20-173 nomad[76334]:     2023-03-21T14:20:00.215Z [INFO]  client.plugin: starting plugin manager: plugin-type=device
Mar 21 14:20:00 ip-10-10-20-173 nomad[76334]:     2023-03-21T14:20:00.227Z [ERROR] client: error discovering nomad servers:
Mar 21 14:20:00 ip-10-10-20-173 nomad[76334]:   error=
Mar 21 14:20:00 ip-10-10-20-173 nomad[76334]:   | 9 errors occurred:
Mar 21 14:20:00 ip-10-10-20-173 nomad[76334]:   | \t* address 2600:(redacted):7629:4647: too many colons in address
Mar 21 14:20:00 ip-10-10-20-173 nomad[76334]:   | \t* address 2600(redacted):1943:4647: too many colons in address
```
Note the lack of square brackets around the ipv6 addresses.

### Reproduction steps
* Configure a Nomad cluster to advertise ipv6 addresses with a `consul` block
* Install Nomad v1.5.1 on a linux server, and configure it to autodiscover the cluster with Consul
* Watch the agent fail to join the cluster with the errors pasted above

**Comment by tgross** [comment]
Hi @protochron! I'm pretty sure this was broken by https://github.com/hashicorp/nomad/pull/16217 which we shipped in Nomad 1.5.1. There's a line where we parsed out the serf tags like this `addr, err := net.ResolveTCPAddr("tcp", fmt.Sprintf("%s:%s", addrTag, portTag))` which is incorrect for IPv6 :facepalm:  We should have done this: `addr, err := net.ResolveTCPAddr("tcp", net.JoinHostPort(addrstr, port))` instead. (Well, _I_ should have done that instead. :blush: )

Fortunately that entire new bit of code was removed in https://github.com/hashicorp/nomad/pull/16490 which will be shipping very soon in Nomad 1.5.2.
